_blog/2018/v1alpha3-routing.md

```diff
@@ -0,0 +1,347 @@
+kind: DestinationRule
+metadata:
+  name: reviews
+spec:
+  host: reviews
+  trafficPolicy:
+    loadBalancer:
+      simple: RANDOM
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+  - name: v2
+    labels:
+      version: v2
+    trafficPolicy:
+      loadBalancer:
+        simple: ROUND_ROBIN
+  - name: v3
+    labels:
+      version: v3
+```
+
+Notice that, unlike `DestinationPolicy`, multiple policies (e.g., default and v2-specific) are specified in a single
+`DestinationRule` configuration.
+
+### ExternalService
+
+`ExternalService` is used to add additional entries into the service registry that Istio maintains internally. This allows
+one to model traffic to external dependencies of the mesh such as APIs consumed from the web or traffic to services in
+legacy infrastructure.
+
+Replacing `EgressRule` from the previous API, `ExternalService` provides several significant improvements, such as:
+
+1. Individual service ports and service discovery mode are now configurable.
+2. One or more (different) external endpoints can be configured to implement a “virtual” external service.
+3. Secure HTTP services (automatic TLS upgrade) can now be accessed using standard https (e.g., `https://secureservice.com/`
+   instead of `http://secureservice.com:443/`.
+4. Multiple CIDR subsets can now be included in a single `ExternalService` configuration.
+
+Because an `ExternalService` configuration simply adds an external destination to the internal service registry, it can be
+used in conjunction with a `VirtualService` and/or `DestinationRule`, just like any other service in the registry.
+
+### Gateway
+
+It all started with ingress. The Istio ingress feature inherited the Kubernetes `Ingress` resource model for expediency, but
+unfortunately it is not able to express all of Istio's routing capabilities. The `Ingress` APIs are inadequate to model
+ingress traffic for several key Istio use-cases such as:
+
+* Split traffic across different versions of a service based on labelling
+* Route traffic based on complex matching criteria on headers and other properties of the call.
+* Apply transformations such as rewrites, header mutations, etc.
+* Define resilience features such as retry & circuit breaking behavior
+* Load balance TCP traffic
+
+Another problem with the `Ingress` specification is that it tries to draw a common denominator across different L7 proxies.
+This results in a least common denominator that uses the most basic HTTP routing and discards every other feature of the
+proxy, which end up in annotations that are not portable.
+
+So the `Gateway` effort started. Istio `Gateway` overcomes the `Ingress` shortcomings by separating the L4-L6 spec from L7.
+It only configures the L4-L6 functions (e.g., ports to expose, TLS configuration) that are uniformly implemented by all good
+L7 proxies. Users can then use standard Istio route rules to control HTTP requests & TCP traffic entering a `Gateway` by
```

Comment:
Shouldn't this be referring to `DestinationRule`? If I understand correctly `RouteRule` is on deprecation track. It currently is a well-adopted descriptor and to avoid confusion should not be used when referring to the combination of `Gateway`, `DestinationRule`, and `VirtualService`. In the future, I am sure, the community will call it that all on _our_ own.